security/cmd/chiron/main.go

```diff
@@ -0,0 +1,313 @@
+	// In prototype, only one is supported.
+	mutatingWebhookServiceNames string
+	// The ports of the services of mutating webhooks to manage
+	// In prototype, only one is supported.
+	mutatingWebhookServicePorts string
+	// The names of the valitatingWebhookConfiguration to manage
+	// In prototype, only one is supported.
+	validatingWebhookConfigNames string
+	// The names of the services of validating webhooks to manage
+	// In prototype, only one is supported.
+	validatingWebhookServiceNames string
+	// The ports of the services of validating webhooks to manage
+	// In prototype, only one is supported.
+	validatingWebhookServicePorts string
+
+	// The minimum grace period for cert rotation.
+	certMinGracePeriod time.Duration
+
+	logOptions *log.Options
+	// Currently, no topic is registered for ctrlz yet
+	ctrlzOptions *ctrlz.Options
+
+	// The length of certificate rotation grace period, configured as the ratio of the certificate TTL.
+	// If certGracePeriodRatio is 0.2, and cert TTL is 24 hours, then the rotation will happen
+	// after 24*(1-0.2) hours since the cert is issued.
+	certGracePeriodRatio float32
+
+	// Whether enable the webhook controller
+	enableController bool
+
+	// Whether delete webhookconfigurations upon exit
+	deleteWebhookConfigurationsOnExit bool
+}
+
+func init() {
+	flags := rootCmd.Flags()
+
+	flags.BoolVar(&opts.enableController, "enable-controller", false, "Specifies whether enabling "+
+		"Istio Webhook Controller.")
+	flags.BoolVar(&opts.deleteWebhookConfigurationsOnExit, "delete-webhook-configurations-on-exit", false, "Specifies whether deleting "+
+		"webhookconfigurations when Istio Webhook Controller exits.")
+	flags.StringVar(&opts.certificateNamespace, "certificate-namespace", "istio-system",
+		"The namespace of the webhook certificates.")
+
+	flags.StringVar(&opts.kubeConfigFile, "kube-config", "",
+		"Specifies path to kubeconfig file. This must be specified when not running inside a Kubernetes pod.")
+
+	// Specifies the file path to k8s CA certificate.
+	// The default value is configured based on https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/:
+	// the CA certificate bundle is automatically mounted into pods using the default
+	// service account at the path /var/run/secrets/kubernetes.io/serviceaccount/ca.crt.
+	flags.StringVar(&opts.k8sCaCertFile, "k8s-ca-cert-file", "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+		"Specifies the file path to k8s CA certificate.")
+
+	// Certificate issuance configuration.
+	flags.Float32Var(&opts.certGracePeriodRatio, "cert-grace-period-ratio",
+		cmd.DefaultWorkloadCertGracePeriodRatio, "The certificate rotation grace period, as a ratio of the "+
+			"certificate TTL.")
+	flags.DurationVar(&opts.certMinGracePeriod, "cert-min-grace-period",
+		cmd.DefaultWorkloadMinCertGracePeriod, "The minimum certificate rotation grace period.")
+
+	flags.StringVar(&opts.webhookConfigFiles, "webhook-config-files",
```

Comment:
Add a TODO. We want to load the config templates from a CRD instead of local file.

Reply:
Done.